### PR TITLE
Notice thrown in bundle edit form after removing last group

### PR DESCRIPTION
**Pull request by pfrenssen**
I'm getting a notice in the bundle edit form after adding a single group type and then removing the group type again:

> Notice: Undefined index: rdf_entity in Drupal\og_ui\BundleFormAlter->addGroupContent() (line 173 of modules/contrib/og/og_ui/src/BundleFormAlter.php).

This happens because when the last group is removed for a given entity type in `GroupManager::removeGroup()` an empty array container is left behind. This fools the calling code in thinking that the entity type still has some groups in it.

### CODE REVIEW

File: src/GroupManager.php [hunk 1]
```diff
@@ -110,6 +110,11 @@ public function removeGroup($entity_type_id, $bundle_id) {
 
       if ($search_key !== FALSE) {
         unset($groups[$entity_type_id][$search_key]);
+        // If the last group of this entity type is removed, also remove the
+        // empty container.
+        if (empty($groups[$entity_type_id])) {
```
Comment (amitaibu) [inline src/GroupManager.php]: this seems a bit to delicate -- which code is being "fooled" by the empty container?

**Comment by pfrenssen** [comment]
The fooling happens in `BundleFormAlter::addGroupContent()` in this section (I replaced the comments):

```
foreach (Og::groupManager()->getAllGroupBundles() as $entity_type => $bundles) {

  // $target_types are being set, even if the $entity_type has no bundles, i.e. if it
  // is an empty container.
  $target_types[$entity_type] = \Drupal::entityTypeManager()->getDefinition($entity_type)->getLabel();

  // This will be skipped since there are no bundles. $target_bundles won't be set.
  foreach ($bundles as $bundle) {
    $target_bundles[$entity_type][$bundle] = $bundle_info[$bundle]['label'];
  }
}

// This switch thinks that there are target types, since one of the $entity_type keys
// was set, even though it was only an empty container.
if ($target_types) {
  // ...
  $form['og']['og_target_bundles'] = [
    // ...
    // And here finally the notice is thrown, $target_bundles was never
    // populated.
    '#options' => $target_bundles[$target_type_default],
    // ...
  ];
}
```

Fixing it here is ugly, we would need to check if there are any bundles before populating the `$target_types`. I first fixed it like this, but then I realized that I would need to check all instances where this is called to see if they deal with empty containers correctly. And not only in the current code but in all future code that relies on the group bundles as well. It just makes more sense to fix it in the method so the config always returns valid, populated data.

**Comment by pfrenssen** [comment]
I changed it to use `array_filter()` instead to clean up orphaned array containers, this looks a lot more sensible :)

File: src/GroupManager.php [hunk 2]
```diff
@@ -114,6 +114,9 @@ public function removeGroup($entity_type_id, $bundle_id) {
         unset($groups[$entity_type_id][$search_key]);
       }
 
+      // Clean up entity types that have become empty.
+      $groups = array_filter($groups);
```
Comment (amitaibu) [inline src/GroupManager.php:118]: 👍

**Pull request merged**